src/core/utils.js

```diff
@@ -137,8 +138,18 @@ const utils = {
         return results;
     },
 
-    urlToContact: function (str) {
-        return kadUtils.parseContactURL(str);
+    urlToContact: function (addr) {
+        const {protocol, hostname, port, hash} = url.parse(addr);
+        const contact = [
+            (hash ? hash.substr(1) : null) ||
+            Buffer.alloc(constants.B / 8).fill(0).toString('hex'),
```

Comment:
I think this function may fail as long as the `constants` variable is undefined.

But basically I propose to remove this `urlToContract` function completely. It is not used anywhere in the code and only makes sense in context of p2p communication within Kademlia network which we decided to give up on.

Reply:
nice